network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/integration/NetPStateCollector.kt

```diff
@@ -20,17 +20,33 @@ import com.duckduckgo.di.scopes.ActivityScope
 import com.duckduckgo.mobile.android.vpn.VpnFeaturesRegistry
 import com.duckduckgo.mobile.android.vpn.state.VpnStateCollectorPlugin
 import com.duckduckgo.networkprotection.impl.NetPVpnFeature
+import com.duckduckgo.networkprotection.impl.connectionclass.ConnectionQualityStore
+import com.duckduckgo.networkprotection.impl.connectionclass.asConnectionQuality
+import com.duckduckgo.networkprotection.impl.store.NetworkProtectionRepository
+import com.duckduckgo.networkprotection.store.NetPExclusionListRepository
 import com.squareup.anvil.annotations.ContributesMultibinding
 import javax.inject.Inject
 import org.json.JSONObject
 
 @ContributesMultibinding(ActivityScope::class)
 class NetPStateCollector @Inject constructor(
     private val vpnFeaturesRegistry: VpnFeaturesRegistry,
+    private val netpRepository: NetworkProtectionRepository,
+    private val netPExclusionListRepository: NetPExclusionListRepository,
+    private val connectionQualityStore: ConnectionQualityStore,
 ) : VpnStateCollectorPlugin {
     override suspend fun collectVpnRelatedState(appPackageId: String?): JSONObject {
+        val isNetpRunning = vpnFeaturesRegistry.isFeatureRunning(NetPVpnFeature.NETP_VPN)
         return JSONObject().apply {
-            put("enabled", vpnFeaturesRegistry.isFeatureRunning(NetPVpnFeature.NETP_VPN))
+            put("enabled", isNetpRunning)
+            if (isNetpRunning) {
+                appPackageId?.let {
+                    put("netpReportedAppProtected", !netPExclusionListRepository.getExcludedAppPackages().contains(it))
```

Comment:
Rename to `reportedAppProtected` (?) netp seems redundant as it is already inside `netState`
 
```suggestion
                    put("reportedAppProtected", !netPExclusionListRepository.getExcludedAppPackages().contains(it))
```